cosmos/client.go

```diff
@@ -173,7 +167,48 @@ func (c *Client) Stream(ctx context.Context, query string) (chan hash.Hash, chan
 	return hashC, errC, nil
 }
 
-// GetAccount returns the keybase's account.
-func (c *Client) GetAccount() (keys.Info, error) {
-	return c.kb.Get(c.accName)
+// GetAccount returns the local account.
+func (c *Client) GetAccount() (auth.Account, error) {
+	c.getAccountMutex.Lock()
+	defer c.getAccountMutex.Unlock()
+	if c.acc == nil {
+		accKb, err := c.kb.Get(c.accName)
+		if err != nil {
+			return nil, err
+		}
+		c.acc = auth.NewBaseAccount(
+			accKb.GetAddress(),
+			nil,
+			accKb.GetPubKey(),
+			AccNumber,
+			0,
+		)
+	}
```

Comment:
This could be moved to a helper method.

Reply:
Yes you're right. We actually plan to move all the account with signing logic to a dedicated struct in a future PR.

Reply:
issue created: <https://github.com/mesg-foundation/engine/issues/1565>